pypsi/shell.py

```diff
@@ -194,6 +194,63 @@ def on_tokenize(self, tokens, origin):
                 break
         return tokens
 
+    def include(self, file):
+        '''
+        Read commands from a file and execute them line by line
+        :param file file: File object to read commands from
+        :return int: 0 if error free; 1 if an error occurred
+        '''
+        rc = 1
+        eof = False
+
+        # set STDIN to the file
+        stdin = sys.stdin._get_target()
+        sys.stdin._proxy(file)
+
+        try:
+            while self.running and not eof:
+                try:
```

Comment:
I think you can wrap the input and execution in a single try...except to catch KeyboardInterrupt, EOFError, and SystemExit only once.